packages/bitcore-node/src/providers/chain-state/evm/models/block.ts

```diff
@@ -189,62 +189,47 @@ export class EVMBlockModel extends BaseBlock<IEVMBlock> {
     return new Promise(async (resolve, reject) => {
       let timeout;
       try {
-        const maxBlock = await self.collection.findOne(
-          {
-            chain,
-            network,
-            // height and processed are here to ensure the `chain_1_network_1_processed_1_height_-1` index is hit
-            height: { $exists: true },
-            $or: [{ processed: { $exists: true } }, { processed: { $exists: false } }]
-          },
-          {
-            // height all that matters in the sort. The rest is there to ensure the `chain_1_network_1_processed_1_height_-1` index is hit
-            sort: { height: -1, chain: 1, network: 1, processed: 1 },
-            projection: { height: 1 }
-          }
-        );
-        if (!maxBlock) {
-          return resolve([]);
-        }
-
         const stream = self.collection
           .find({
             chain,
             network,
-            height: { $gte: startHeight },
-            // processed is here to ensure the `chain_1_network_1_processed_1_height_-1` index is hit
-            $or: [{ processed: { $exists: true } }, { processed: { $exists: false } }]
+            height: { $gte: startHeight }
           })
-          .sort({ chain: 1, network: 1, height: 1, processed: 1 }) // height is the only sort that matters
+          .sort({ chain: 1, network: 1, processed: 1, height: -1 }) // guarantee index use by using this sort
           .addCursorFlag('noCursorTimeout', true);
 
-        const maxHeight = maxBlock.height;
         let block = (await stream.next()) as IEVMBlock;
+        const maxBlock = block;
+        if (!maxBlock) {
+          return resolve([]);
+        }
+        const maxHeight = maxBlock.height;
         let prevBlock: IEVMBlock | undefined;
         const outOfSync: number[] = [];
         timeout = setInterval(
           () => logger.info(`${chain}:${network} Block verification height: ${block.height}`),
```

Comment:
Since block.height is descending now, let's change this to:
```
        timeout = setInterval(
          () => logger.info(`${chain}:${network} Block verification progress: ${((maxHeight - block.height) / (maxHeight - startHeight) * 100).toFixed(1)}%`),
          1000 * 2
        );
```

Reply:
Good idea. I implemented that.